micrometer-core/src/main/java/io/micrometer/core/instrument/push/PushMeterRegistry.java

```diff
@@ -60,6 +62,7 @@ protected PushMeterRegistry(PushRegistryConfig config, Clock clock) {
     // VisibleForTesting
     void publishSafely() {
         if (this.publishing.compareAndSet(false, true)) {
+            this.lastScheduledPublishStartTime = clock.wallTime();
```

Comment:
I don't think it's a problem with the current implementation, but it feels like it would be more correct to set this directly from the scheduled task rather than in this method. After all, this method *could* be called from outside scheduling. I'm thinking updating the scheduling to something like:

```java
scheduledExecutorService.scheduleAtFixedRate(() -> {
    this.lastScheduledPublishStartTime = clock.wallTime();
    publishSafely();
}, initialDelayMillis, stepMillis, TimeUnit.MILLISECONDS);
```

What do you think?

Reply:
Yeah, considering everything identified so far it seems like it will be better to leave it as is for now.